components/map/basic/src/index.js

```diff
@@ -0,0 +1,257 @@
+    this.setMapEventDefinition(props)
+    this.mapInstance = undefined
+    this.isHeatmapVisible = false
+    this.isSatelliteView = false
+  }
+
+  setMapEventDefinition () {
+    this.mapEventList = [
+      {
+        name: 'leaflet_map_click',
+        handleFunction: (evt) => this.props.onMapClick(evt.detail)
+      },
+      {
+        name: 'leaflet_map_dragend',
+        handleFunction: (evt) => this.props.onMapDragEnd(evt.detail)
+      },
+      {
+        name: 'leaflet_map_loaded',
+        handleFunction: (evt) => this.props.onMapLoad(evt.detail)
+      },
+      {
+        name: 'leaflet_map_zoomend',
+        handleFunction: (evt) => this.props.onMapZoomEnd(evt.detail)
+      },
+      {
+        name: 'leaflet_map_layer_normal',
+        handleFunction: (evt) => this.props.onNormalView(evt.detail)
+      },
+      {
+        name: 'leaflet_map_poiclick',
+        handleFunction: (evt) => this.props.onPoiClick(evt.detail)
+      },
+      {
+        name: 'leaflet_map_poimouseout',
+        handleFunction: () => this.props.onPoiMouseOut()
+      },
+      {
+        name: 'leaflet_map_poimouseover',
+        handleFunction: (evt) => this.props.onPoiMouseOver(evt.detail)
+      },
+      {
+        name: 'leaflet_map_layer_satellite',
+        handleFunction: (evt) => this.props.onSatelliteView(evt.detail)
+      }
+    ]
+  }
+
+  getMapConfig () {
+    return {
+      id: this.props.id,
+      heatMapUrl: this.props.heatMapUrl,
+      latitude: this.props.center[0],
+      longitude: this.props.center[1],
+      literals: this.props.literals,
+      maxZoom: this.props.maxZoom,
+      minZoom: this.props.minZoom,
+      polygons: this.props.polygons,
+      showHeatmap: this.props.showHeatmap,
+      showSatelliteView: this.props.showSatelliteView,
+      mapViewModes: this.props.mapViewModes,
+      selectedMapViewMode: this.props.selectedMapViewMode,
+      zoomControl: this.props.zoomable,
+      zoomControlPosition: this.props.zoomControlPosition,
+      icons: this.props.icons,
+      enableViewMenu: this.props.enableViewMenu,
+      dragging: this.props.isInteractable,
+      zoom: this.props.zoom,
+      appId: this.props.appId,
+      appCode: this.props.appCode
+    }
+  }
+
+  subscribeToMapEvents () {
+    this.mapEventList.forEach(mapEvent => this.mapDOMInstance.addEventListener(mapEvent.name, mapEvent.handleFunction))
+  }
+
+  unsubscribeFromMapEvents () {
+    this.mapEventList.forEach(mapEvent => this.mapDOMInstance.removeEventListener(mapEvent.name, mapEvent.handleFunction))
+  }
+
+  checkWhichViewShouldBeDisplayed (showSatelliteView) {
+    if (showSatelliteView && !this.isSatelliteView) {
+      this.isSatelliteView = true
+      this.mapInstance.setView(mapViewModes.SATELLITE)
+    } else if (!showSatelliteView && this.isSatelliteView) {
+      this.isSatelliteView = false
+      this.mapInstance.setView(mapViewModes.NORMAL)
+    }
+  }
+
+  checkIfHeatMapShouldBeDisplayed (showHeatmap, url) {
```

Comment:
You're not checking, you're display the heapmap... maybe `toggleHeatMap`of  `showHeatMap`

Reply:
Same here. Is not a pure toggling function. I thought the same on my first function view but later I realised that is more a check&execute function than a pure toggle one.